### PR TITLE
Separate config for Bahmni diagnoses

### PR DESCRIPTION
**Pull request by kaapstorm**
##### SUMMARY
This PR separates configuration for Bahmni diagnoses from configuration for OpenMRS Observations.

##### FEATURE FLAG
OpenMRS integration

##### PRODUCT DESCRIPTION
Configuration for Bahmni diagnoses is now separated from configuration for OpenMRS Observations. They are similar and can be configured the same way, but they are not the same and this change is to avoid potential confusion.

### CODE REVIEW

**Review by esoergel — APPROVED** [review]

File: corehq/motech/openmrs/atom_feed.py [hunk 1]
```diff
@@ -359,6 +359,18 @@ def import_encounter(repeater, encounter_uuid):
...
+            repeater.diagnosis_mappings,
+            case_id,
+            case_type,
+            default_owner_id,
+        )
+        deep_update(case_block_kwargs, more_kwargs)
+        case_blocks.extend(more_case_blocks)
+        # O/ ... ... ... start snip
+        # TODO: Remove this after deploy, once existing configs have
+        #       been moved to repeater.bahmni_diagnoses
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply corehq/motech/openmrs/atom_feed.py]: AFAIK, there are only two domains importing Bahmni diagnoses on prod. (The feature is quite new.) I was planning to iterate domains with OpenMRS integration installed, checking their configs, and moving Bahmni diagnoses into their own section manually if there are only a few.

[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/motech/openmrs/atom_feed.py]: okay, sounds reasonable

**Review by esoergel — APPROVED** [review]
What's going on with the commit history?  Your last two commits are merge commits, but it looks like they also add changes.

**Comment by kaapstorm** [comment]
Both merges were to resolve conflicts from changes in the same area. The first merge resolves conflicts introduced by PR #25862, and the second by PR #25968.

**Review by esoergel — APPROVED** [review]

**Pull request merged**